### PR TITLE
Add Calculator Components

### PR DESCRIPTION
**Pull request by Kamba56**
### This pull request contains the following changes;

- Added the components of the calculator
- Styled the components to look like the calculator design

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <input type="text" placeholder={0} className="calc-input" />
+    <div className="btn-contain">
+      {buttonArr.map((btn) => (
+        <Btn key={btn} btnName={btn} />
+      ))}
+    </div>
+  </div>
+);
+
+export default Calculator;
```
Comment (ahmetbozaci) [inline src/components/calculator.js]: - [ ] Nice to use functional components but in this milestone, you should use class based components. You will refactor it in next milestones[Requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md#:~:text=Use%20class%20based%20components%20(All%20components%20should%20be%20class%2Dbased).)

File: src/App.js [hunk 2]
```diff
@@ -1,27 +1,10 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+const FullCalculator = () => (
```
Comment (ahmetbozaci) [inline src/App.js]: - [ ] Nice to use functional components but in this milestone, you should use class based components. You will refactor it in next milestones[Requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md#:~:text=Use%20class%20based%20components%20(All%20components%20should%20be%20class%2Dbased).)

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @Kamba56  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- You follow the design.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: src/App.js (hunk 2 above)
Comment (Kamba56) [reply src/App.js]: Hello @ahmetbozaci
Thanks for the review.
eslint flags an error when I use class components that's why I used functional components.
Is there a solution for that?

Comment (ahmetbozaci) [reply src/App.js]: @Kamba56  I don't remember the error name. Kindly check the error in google. It has a solution for that. If you send me the error name ı can send the link.

**Comment by Kamba56** [comment]
@ahmetbozaci 
I am getting a useless constructor error.
Should I remove the contructors?

**Comment by ahmetbozaci** [comment]
> I am getting a useless constructor error.

You can fix it in writing class with React.PureComponent

File: src/App.js [hunk 3]
```diff
@@ -1,27 +1,14 @@
...
-    </div>
-  );
+class FullCalculator extends React.component {
+  render() {
+    return (
+      <section className="calc-section">
+        <Calculator />
+      </section>
+    );
+  }
```
Comment (brytebee) [inline src/App.js]: - [ ] **Required:** Your app throws this console error and renders an empty page. Kindly fix and request another review.

image

**Comment by Kamba56** [comment]
> > I am getting a useless constructor error.
> 
> You can fix it in writing class with React.PureComponent

Thank you

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Kamba56👋, 

Great work on making the changes 👏 👏 
You've done well implementing some of the requested changes requested by the previous review, but there are still some which aren't addressed yet.


# Suggested changes
_Check the comments under the review._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by medaminedev66 — APPROVED** [review]
Hi @Kamba56 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- Your web app runs without errors.

### Optional suggestions
- There is still some boilerplate code and images which need to be deleted from your repo. It would be better to remove all unused files to keep your repo clean.

image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**